fairmq/sdk/Topology.h

```diff
@@ -61,30 +63,45 @@ struct DeviceStatus
 {
     bool initialized;
     DeviceState state;
+    DDSTask::Id taskId;
+    DDSCollection::Id collectionId;
 };
 
-using TopologyState = std::unordered_map<DDSTask::Id, DeviceStatus>;
+using TopologyState = std::vector<DeviceStatus>;
+using TopologyStateByTask = std::unordered_map<DDSTask::Id, DeviceStatus>;
+using TopologyStateByCollection = std::unordered_map<DDSCollection::Id, std::vector<DeviceStatus>>;
 using TopologyTransition = fair::mq::Transition;
 
 inline DeviceState AggregateState(const TopologyState& topologyState)
 {
-    DeviceState first = topologyState.begin()->second.state;
+    DeviceState first = topologyState.begin()->state;
 
     if (std::all_of(topologyState.cbegin(), topologyState.cend(), [&](TopologyState::value_type i) {
-            return i.second.state == first;
+            return i.state == first;
         })) {
         return first;
     }
 
     throw MixedStateError("State is not uniform");
-
 }
 
 inline bool StateEqualsTo(const TopologyState& topologyState, DeviceState state)
 {
     return AggregateState(topologyState) == state;
 }
 
+inline TopologyStateByCollection GroupByCollectionId(const TopologyState& topologyState)
```

Comment:
Let's also add the analog `inline TopologyStateByTask GroupByTaskId(const TopologyState& topologyState)` helper.